subsys/bluetooth/services/fast_pair/fp_crypto/include/fp_crypto_pn.h

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef _FP_CRYPTO_PN_H_
+#define _FP_CRYPTO_PN_H_
+
+#include <sys/types.h>
+
+/**
+ * @defgroup fp_crypto_pn Fast Pair crypto Personalized Name
+ * @brief Internal API for Fast Pair crypto that is used by Personalized Name extension.
+ *
+ * @{
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** Length of nonce (64 bits = 8 bytes). */
+#define FP_CRYPTO_PN_NONCE_LEN	8U
+
+/** Encrypt data using AES-CTR.
+ *
+ * @param[out] out Buffer to receive encrypted data.
+ * @param[in] in Plaintext data.
+ * @param[in] data_len Length of input and output data (in bytes).
+ * @param[in] key 128-bit (16-byte) AES key.
+ * @param[in] nonce 64-bit (8-byte) nonce.
+ *
+ * @return 0 If the operation was successful. Otherwise, a (negative) error code is returned.
+ */
+int fp_crypto_aes_ctr_encrypt(uint8_t *out, const uint8_t *in, size_t data_len, const uint8_t *key,
+			      const uint8_t *nonce);
+
+/** Decrypt data using AES-CTR.
+ *
+ * @param[out] out Buffer to receive plaintext data.
+ * @param[in] in Ciphertext data.
+ * @param[in] data_len Length of input and output data (in bytes).
+ * @param[in] key 128-bit (16-byte) AES key.
+ * @param[in] nonce 64-bit (8-byte) nonce used to encrypt data.
+ *
+ * @return 0 If the operation was successful. Otherwise, a (negative) error code is returned.
+ */
+int fp_crypto_aes_ctr_decrypt(uint8_t *out, const uint8_t *in, size_t data_len, const uint8_t *key,
+			      const uint8_t *nonce);
+
+/** Generate HMAC-SHA256.
+ *
+ * @param[out] out 256-bit (32-byte) buffer to receive hashed result.
+ * @param[in] in Input data.
+ * @param[in] data_len Length of input data.
```

Comment:
Maybe we could somehow emphasize that input data is the actual data + nonce to prevent confusion?

Reply:
This is just a general function to calculate HMAC-SHA256. It calculates hash from some bytes. Fast Pair personalized name extension just uses this function in some specific case. User can use `fp_crypto_additional_data_encode` that will calculate hash for him. This HMAC-SHA256 function may be useful in future. Also I want to keep it to be able to easily test it.